File: javascript/bob/bob.js
var Bob = function() {};

Bob.prototype.hey = function(input) {
  if (input === input.toUpperCase()  && input != input.toLowerCase()) {
    return 'Whoa, chill out!';
  } if (input.endsWith("?")) {
    return 'Sure.';
  } if (input.replace(/\s/g, "") === "") {
    return 'Fine. Be that way!';
  }
    return "Whatever.";
};

module.exports = Bob;

File: python/bob/bob.py
import re

def hey(what):
    what = what.strip()
    if what.isupper() and what.lower() != what:
        return 'Whoa, chill out!'
    elif what.endswith('?'):
        return 'Sure.'
    elif len(what) == 0:
        return 'Fine. Be that way!'
    else:
        return 'Whatever.'
